lib/engines/content_object_store/app/services/content_object_store/update_edition_service.rb

```diff
@@ -31,8 +31,20 @@ def call(edition_params)
   private
 
     def create_new_content_block_edition_for_document(edition_params:)
+      validate_existing_edition_params = edition_params.except(:creator)
+      if validate_existing_edition_params.key?(:content_block_document_attributes)
+        validate_existing_edition_params[:content_block_document_attributes] = edition_params[:content_block_document_attributes]
+          .except(:block_type)
+          .merge(id: @original_content_block_edition.document.id)
+      end
```

Comment:
not sure I totally understand what lines 34-39 are doing... is it ensuring that the Document ID is there, and that `creator` and `block_type` are removed from the fields updated? Why do we have to remove them, because they shouldn't be changed in an update?

Reply:
This is indeed a bit sticky. 

Since we have to call `valid?` on the _original_ object we need to be careful about how we send it information. We want the validation feedback but we need to avoid setter violations and prevent it creating a brand new `Document`.

- `creator` has to be removed since this it not an updatable field and so would falsely fail the validation every time
- `block_type` also cannot be updated since it's `attr_readonly`
- we add Document `id` to associate the records again, preventing it creating a new `Document` object with the new`title` value

Reply:
I've lifted this responsibilities and given it a name and some comments as I think it's a rare case where they might be useful.